### PR TITLE
[3.3.x] Bump version for the 3.3 prerelease

### PR DESCRIPTION
**Pull request by jtpio**
## References

As mentioned in https://github.com/jupyterlab/jupyterlab/issues/11643#issuecomment-1007493038

This should fix CI issues on the `3.3.x` branch as noticed in https://github.com/jupyterlab/jupyterlab/pull/11804

## Code changes

- Bump the JS versions to `3.3.0-alpha.15`
- Keep `@jupyterlab/ui-components` at `3.3.0-alpha.14` because it was bumped independently as part of https://github.com/jupyterlab/jupyterlab/pull/11173
- Bump the Python version to `3.3.0a0`

This bump is done manually because some of the JS versions have already been used and released (during the `4.0.0ax` series) on `master` when the major version bump had not happened yet. For example: 

https://github.com/jupyterlab/jupyterlab/blob/d1244eaa11eada494d98826e16ca42b31b9476f5/packages/application/package.json

## User-facing changes

The first 3.3 prerelease will then be `3.3.0a1` since the releaser will bump the version during the release process

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by jtpio** [comment]
The failing check is expected (more than 100 labels applied).

**Comment by jtpio** [comment]
Pending the remaining CI checks, this should now be good to go.

Ideally we would have done the bump and release with the releaser directly. But because of some `3.3.0-alpha.x` packages already used and published (detailed in https://github.com/jupyterlab/jupyterlab/issues/11643#issuecomment-1007493038), the script would have bumped versions from `3.2.x` to `3.3.0-alpha.0`.

File: jupyterlab/staging/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@jupyterlab/application-top",
-  "version": "3.2.5",
+  "version": "3.3.0-alpha.15",
```
Comment (jtpio) [inline jupyterlab/staging/package.json:3]: Here we manually update the `staging` files just to be able to realign the versions.

Normally these files should not be updated in regular PRs: https://github.com/jupyterlab/jupyterlab/issues/7237

**Comment by jtpio** [comment]
Let's merge to make sure this fixes the CI. 

We can rebase some of the opened backport PRs targeting `3.3.x` right after.

**Review by fcollonval — APPROVED** [review]
Thanks a lot @jtpio

**Pull request merged**